src/layout/mod.rs

```diff
@@ -3950,6 +3971,27 @@ mod tests {
                         op.apply(&mut layout);
                     }
 
+                    first.apply(&mut layout);
+                    layout.verify_invariants();
+                    second.apply(&mut layout);
+                    layout.verify_invariants();
+                    third.apply(&mut layout);
+                    layout.verify_invariants();
+                }
+                // same test with empty-workspace-above-first set to true
```

Comment:
Not sure I want to add this since it doubles the test time (which is already the longest I believe). Let's let proptest catch these issues, together with manual test cases.

Reply:
ok removed it again